src/state/routing/uraSlice.ts

```diff
@@ -0,0 +1,122 @@
+import { createApi, fetchBaseQuery, FetchBaseQueryError } from '@reduxjs/toolkit/query/react'
+import { Protocol } from '@uniswap/router-sdk'
+import { getClientSideQuote } from 'lib/hooks/routing/clientSideSmartOrderRouter'
+import ms from 'ms.macro'
+import { trace } from 'tracing/trace'
+
+import { GetQuoteArgs, INTERNAL_ROUTER_PREFERENCE_PRICE, RouterPreference } from './slice'
+import { QuoteState, TradeResult, URAQuoteData } from './types'
+import { getRouter, isAPIAcceptedRouterPreference, isExactInput, transformRoutesToTrade } from './utils'
+
+const CLIENT_PARAMS = {
+  protocols: [Protocol.V2, Protocol.V3, Protocol.MIXED],
+}
+
+// routing API quote query params: https://github.com/Uniswap/routing-api/blob/main/lib/handlers/quote/schema/quote-schema.ts
+const CLASSIC_SWAP_QUERY_PARAMS = {
+  ...CLIENT_PARAMS,
+  routingType: 'CLASSIC',
+}
+
+export const uraRoutingApi = createApi({
+  reducerPath: 'ura-routing',
+  baseQuery: fetchBaseQuery({
+    baseUrl: 'https://4uemel8n4g.execute-api.us-east-2.amazonaws.com/prod',
```

Comment:
awaiting final URL for here

Reply:
is this something we usually put in the .env file?

Reply:
we can, we don't typically switch api URLs for staging though so dont know if that would make sense